### ISSUE TITLE
Bug: [@typescript-eslint/return-await] false positives in 'always' mode

### ISSUE THREAD

**Issue by liosha** [typescript-eslint/typescript-eslint#12072]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.9.3&fileType=.ts&code=PTAEBMFMDMGdWgJwPYFtSQB4BdKIHYCGANqMQJYBGihiAngLABQ5%2Bui0hAxpKAEqRYAB2T5YkAPKUAVgB4AIoWyEAKnSGQAfKADezUBCWFZK0AF5Qi5Wo2aAFAEoAXKBUBuZgF9mzVu048lka6%2BgjQLrDYiKwA5l4%2BLGx4AbwAwhSQbCFMBlwkxC6O5toACiio5OKyAsKi4lJyVoSamvFMzCCgyADu%2BKBcyFDMxJDY-RlsLunkmdjMA2JjqIQA1pAAgqn55qCEsHT4XKCOLmVolZAKRtpm2no5oIijAK4Ex4TdhORjXBPYAHR5YjERwOf7gIxFTqccjEWBteZ1JarDZbYEAJh2ewORxOoDOFSqTRud1CC0ij0EIjEvAsHy%2BPz%2BgPyjlCT2wrz6T1qNPBkIcoE6kVhpBhcIRTHJyLWm3yAGYsftDsdnPjyhcrsoSdlckjKTzxFjPt9xjM2MzgayHuzOfrqeI%2BcpNc0oWAsEIKFwTdh1LwhLRCOg7LBA7w9hAYIRnsRsAKALTafDIBBfYivSASqWgZYytHEAAsipxKtO6qJ12KOv6eu50bGFjs9JNvzNAKBIIcYIhyldoEiyGi%2BBiZGQ7booAToCTKdh6bZLzetZjbSAA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6aRfWaJ2gQwDufSvnRgA2pETRoAe2iQANFD7whhVAF1wYAL4hdQA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=true

### Repro Code

```TypeScript
interface ResponseObj<DataType> {
  data<T = DataType>(): T;
}

interface Data {
  ff: string
}

interface Client {
  call: () => Promise<ResponseObj<Data>>
}

let client: Client
const makeACall = async (): Promise<Data> => {
  const response = await client.call()
  return response.data() // fails
}
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "@typescript-eslint/return-await": ["error", "always"]
  }
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
  }
}
```

### Expected Result

Rule does not fail on `return response.data()` line, as it returns non-promise value

### Actual Result

Rule fails on `return response.data()` line

**Comment by bradzacher** [comment]
Use your IDE to inspect the type of `.data()` in the failing case you'll quickly see what's going on.

TypeScript is back-filling the generic using the return type of the function - so what you're ending up with is that the return type of `data()` is `Data | PromiseLike<Data>`.

This is why the rule flags - based on the types the expression could be a promise and thus it must be awaited.

This is one of the many reasons that generics that are only used in a return type location are considered an unsafe anti-pattern.
